modules/ballerina-mime/src/main/ballerina/ballerina/mime/natives.bal

```diff
@@ -0,0 +1,293 @@
+@Field {value:"headers: Denote general, request/response and entity related headers. Keys of the header map
+should represent the header name and value will be the 'HeaderValue' struct"}
+@Field {value:"isInMemory: A boolean to represent whether the body of the entity is in memory or in a temporary file"}
+@Field {value:"textData: Contents of the body in string form if the content is of text type"}
+@Field {value:"jsonData: Contents of the body in json form if the content is of json type"}
+@Field {value:"xmlData: Contents of the body in xml form if the content is of xml type"}
+@Field {value:"byteData: Contents of the body as a byte array"}
+@Field {value:"overflowData: If the size of the entity exceeds the limit defined in ballerina service or the client
+connector, contents will be saved in a temporary file and can be accessed through this file handler"}
+@Field {value:"multipartData: Represent body parts(entities) of a multipart entity"}
+@Field {value:"size: Represent the size of the entity"}
+@Field {value:"fileName: Default filename for storing the bodypart, if the receiving agent wishes to store it in an
+external file"}
+@Field {value:"contentDisposition: Indicates how the body part should be presented (inline, attachment or as
+form-data)"}
+@Field {value:"name: Represent the field name in case of 'multipart/form-data'"}
+public struct Entity {
+    MediaType contentType;
+    string contentId;
+    map headers;
+    boolean isInMemory;
+    string textData;
+    json jsonData;
+    xml xmlData;
+    blob byteData;
+    file:File overflowData;
+    Entity[] multipartData;
+    int size;
+    string fileName;
+    Disposition contentDisposition;
+    string name;
+}
+
+@Description {value:"Indicates how the body part should be presented."}
+@Field {value:"INLINE: Is intended to be displayed automatically upon display of the message"}
+@Field {value:"ATTACHMENT: Indicate that the body part is separate from the main body of the top level message"}
+@Field {value:"FORM_DATA: When the media type is multipart/form-data, content disposition will be 'form-data'"}
+public enum Disposition {
+    INLINE,
+    ATTACHMENT,
+    FORM_DATA
+}
+
+@Description {value:"Given an entity, get the text payload, either from the memory or from the file handler."}
+@Param {value:"entity: Represent mime Entity"}
+@Return {value:"return text data"}
+public function getText (Entity entity) (string) {
+    if (entity.isInMemory) {
+        return entity.textData;
+    } else {
+        file:File overFlowData = entity.overflowData;
+        if (overFlowData != null) {
+            var encoding = DEFAULT_CHARSET;
+            TypeCastError castErr;
+            if (entity.contentType != null) {
+                encoding, castErr = (string)entity.contentType.parameters.CHARSET;
+                if (castErr != null) {
+                    encoding = DEFAULT_CHARSET;
+                }
+            }
```

Comment:
The above code block of getting encoding is duplicated over several methods. Can we create a private function for that and use within each get method